### PR TITLE
Media type fix

### PR DESCRIPTION
**Pull request by cabo**
This contains a few editorial fixes (section references, marked-up BCP14 words), and, most importantly, updates to the IANA registrations based on Designated Expert review.

### CODE REVIEW

File: draft-ietf-core-sid-latest.md [hunk 1]
```diff
@@ -555,18 +550,16 @@ This document registers one YANG module in the "YANG Module Names" registry {{RF
...
-| application/yang-data+cbor; id=sid    |                | TBD1 | RFC XXXX  |
-{: align="left"}
+## CoAP Content-Formats Registry
 
-// RFC Ed.: replace TBD1 with assigned IDs and remove this note.
-// RFC Ed.: replace RFC XXXX with this RFC number and remove this note.
+A Content-Format for `application/yang-data+cbor; id=sid` has been
+added to the "CoAP Content-Formats" within the "Constrained RESTful
+Environments (CoRE) Parameters" registry, in
+{{I-D.ietf-core-yang-cbor}}.
```
Comment (ektrah) [inline draft-ietf-core-sid-latest.md:562]: It seems this section is no longer needed then?

File: draft-ietf-core-yang-cbor-latest.md [hunk 2]
```diff
@@ -1681,36 +1716,40 @@ Author/Change controller:
...
 This document adds the following Content-Format to the "CoAP Content-Formats",
-within the "Constrained RESTful Environments (CoRE) Parameters" registry.
-
-| Media Type                             | Content Coding | ID   | Reference |
-| application/yang-data+cbor; id=name    |                | TBD1 | RFC XXXX  |
+within the "Constrained RESTful Environments (CoRE) Parameters"
+registry, where TBD3 comes from the "Expert Review" 0-255 range and
+TBD1 and TBD2 come from the "IETF Review" 256-9999 range.
+
+| Media Type                          | Content Coding | ID   | Reference |
```
Comment (ektrah) [inline draft-ietf-core-yang-cbor-latest.md:1723]: RFC 7252 actually wants this table to have the following labels:

```
| Media type               | Encoding | ID | Reference              |
```

The following labels would be correct, though:

```
| Content Type          | Content Coding | ID | Reference              |
```

**Comment by cabo** [comment]
https://trac.ietf.org/trac/xml2rfc/ticket/639
I knew this would bite me.

File: draft-ietf-core-yang-cbor-latest.md [hunk 3]
```diff
@@ -1681,36 +1716,40 @@ Author/Change controller:
...
+| application/yang-data+cbor          |                | TBD1 | RFC XXXX  |
```
Comment (ektrah) [inline draft-ietf-core-yang-cbor-latest.md:1724]: I think we usually put a `-` as the Content Coding in RFCs to make it obvious that it has been filled out. (The IANA registry then leaves it empty, though.)

File: draft-ietf-core-yang-cbor-latest.md [hunk 4]
```diff
@@ -1681,36 +1716,40 @@ Author/Change controller:
 ## CoAP Content-Formats Registry
 
 This document adds the following Content-Format to the "CoAP Content-Formats",
-within the "Constrained RESTful Environments (CoRE) Parameters" registry.
-
-| Media Type                             | Content Coding | ID   | Reference |
-| application/yang-data+cbor; id=name    |                | TBD1 | RFC XXXX  |
+within the "Constrained RESTful Environments (CoRE) Parameters"
+registry, where TBD3 comes from the "Expert Review" 0-255 range and
+TBD1 and TBD2 come from the "IETF Review" 256-9999 range.
```
Comment (ektrah) [inline draft-ietf-core-yang-cbor-latest.md:1721]: Please don't put the requested ranges in the final RFC. (A note-to-be-removed-by-the-RFC-Editor is fine.)

File: draft-ietf-core-yang-cbor-latest.md [hunk 5]
```diff
@@ -1608,10 +1618,12 @@ CBOR encoding:
 
 # Content-Types {#content-type}
 
-The following Content-Type is defined:
+Based on the media-type `application/yang-data+cbor`, the following
+Content-Types are defined:
```
Comment (ektrah) [inline draft-ietf-core-yang-cbor-latest.md:1622]: Should this section say something about future additions to the set of possible values of the `id` parameter?

File: draft-ietf-core-yang-cbor-latest.md [hunk 6]
```diff
@@ -1620,7 +1632,30 @@ application/yang-data+cbor; id=name:
...
+
+For CBOR YANG documents making use of YANG SID identifiers only:
+
+application/yang-data+cbor; id=sid:
+: This Content-Type represents a CBOR YANG document containing one or multiple data node values.
+  Each data node is identified by its associated SID as defined in {{sid}}.
+
+: FORMAT: CBOR map of SID, instance-value
+
+: The message payload of Content-Type 'application/yang-data+cbor' is encoded using a CBOR map.
```
Comment (ektrah) [inline draft-ietf-core-yang-cbor-latest.md:1646]: It feels a bit odd that a parameter value such as `sid` defines the entire message format from scratch. Wouldn't it be better to specify the CBOR structure once and then describe how the `id` parameter further constrains the CBOR structure in a certain way?

**Review by ektrah — CHANGES_REQUESTED** [review]
I've done a quick review of the CoAP-related IANA Considerations; see below.

File: draft-ietf-core-sid-latest.md (hunk 1 above)
Comment (cabo) [reply draft-ietf-core-sid-latest.md:562]: It is not needed.  However, it is useful for people who wonder about Content-Format registrations.

File: draft-ietf-core-yang-cbor-latest.md (hunk 2 above)
Comment (cabo) [reply draft-ietf-core-yang-cbor-latest.md:1723]: I'll go for the latter then...

File: draft-ietf-core-yang-cbor-latest.md (hunk 3 above)
Comment (cabo) [reply draft-ietf-core-yang-cbor-latest.md:1724]: Fixed.

File: draft-ietf-core-yang-cbor-latest.md (hunk 5 above)
Comment (cabo) [reply draft-ietf-core-yang-cbor-latest.md:1622]: We don't know anything about that future.
What can we say at this point?
Any additional id value would be an update as per section 5.5 of RFC 6383.
(See also the internally inconsistent section 4.3.)

File: draft-ietf-core-yang-cbor-latest.md (hunk 6 above)
Comment (cabo) [reply draft-ietf-core-yang-cbor-latest.md:1646]: Yes, I'm a bit uncomfortable with this section now, too.
So after pushing the above fixes I'll have a go at refactoring it.

File: draft-ietf-core-yang-cbor-latest.md (hunk 5 above)
Comment (ektrah) [reply draft-ietf-core-yang-cbor-latest.md:1622]: > What can we say at this point?

Are future additions possible at all? And, if yes, what's the ABNF for potential parameter values?

That's the two things that come to my mind.

File: draft-ietf-core-sid-latest.md (hunk 1 above)
Comment (ektrah) [reply draft-ietf-core-sid-latest.md:562]: If IANA is happy with a no-op section, I'm happy too 😊

File: draft-ietf-core-yang-cbor-latest.md (hunk 5 above)
Comment (cabo) [reply draft-ietf-core-yang-cbor-latest.md:1622]: OK, had a crystal ball handy.

**Review by ivajloip — APPROVED** [review]
Thanks, that looks good to me! :slightly_smiling_face: I am not sure if you still want to reformulate the content format description and that's why for now I will not merge it.

File: draft-ietf-core-yang-cbor-latest.md [hunk 7]
```diff
@@ -1620,7 +1632,33 @@ application/yang-data+cbor; id=name:
...
+For CBOR YANG documents making use of potentially mixed identifiers:
+
+application/yang-data+cbor:
+: This Content-Type represents a CBOR YANG document containing one or
+  multiple data node values, where each data nodes may be identified either
+  by a name (as in `id=name`) or by a YANG SID (as in `id=sid`).
+
+It is not foreseen at this point that the valid set of values for the
+`id` parameter will extend `name`, `sid`, or being unset; if that does
+happen, any new value is foreseen to be of the form `[a-z]+`.
```
[the remark this answers is not in the document]
Comment (ektrah) [reply draft-ietf-core-yang-cbor-latest.md:1661]: Mhh... [rfc6838](https://datatracker.ietf.org/doc/html/rfc6838#section-4.3) says:

> Parameter names have the syntax as media type names and values:
>
> ```
> parameter-name = restricted-name
> ```
>
> Note that this syntax is somewhat more restrictive than what is allowed by the ABNF in [RFC2045] and amended by [RFC2231].
>
> Parameter names are case-insensitive and no meaning is attached to the order in which they appear.  It is an error for a specific parameter to be specified more than once.
>
> There is no defined syntax for parameter values.  Therefore, registrations MUST specify parameter value syntax.  Additionally, some transports impose restrictions on parameter value syntax, so care needs be taken to limit the use of potentially problematic syntaxes; e.g., pure binary valued parameters, while permitted in some protocols, are best avoided.

[rfc7231](https://datatracker.ietf.org/doc/html/rfc7231#section-3.1.1.1) defines the ABNF for media type parameters as:

```
parameter      = token "=" ( token / quoted-string )
```

And [rfc7230](https://datatracker.ietf.org/doc/html/rfc7230#section-3.2.6) defines the ABNF for tokens as:

```
token          = 1*tchar

tchar          = "!" / "#" / "$" / "%" / "&" / "'" / "*"
               / "+" / "-" / "." / "^" / "_" / "`" / "|" / "~"
               / DIGIT / ALPHA
               ; any VCHAR, except delimiters
```

I think I'd go with `token` to allow for something like a future `id=sid2.0`. (Or maybe simply `token / quoted-string`?)

File: draft-ietf-core-yang-cbor-latest.md (hunk 4 above)
Comment (cabo) [reply draft-ietf-core-yang-cbor-latest.md:1724]: Now says "// RFC Ed.: please replace TBDx with assigned IDs, remove the
requested ranges, and remove this note.\\"

File: draft-ietf-core-yang-cbor-latest.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (cabo) [reply draft-ietf-core-yang-cbor-latest.md]: I think this is overdoing it.  [a-z][a-z0-9]*(-[a-z0-9]+)* is about as far as I think we should be going.  Submitted a cleaned-up version of Section 7.  Should be ready to go now!

**Pull request merged**